### PR TITLE
refactor dbListTables and getTableNames

### PR DESCRIPTION
**Pull request by ablack3**
Refactoring getTableNames so it calls dbListTables and works with multiple drivers (DatabaseConnector, odbc, RPostgres, duckdb)

@schuemie If this looks good I'll add tests for other DBI driver packages.

### CODE REVIEW

File: R/ListTables.R [hunk 1]
```diff
@@ -16,65 +16,114 @@
...
+      DBI::dbListTables(connection, schema = databaseSchema)
+    } else {
+      DBI::dbListTables(connection, database = databaseSchema[1], schema = databaseSchema[2])
     }
-    databaseSchema <- strsplit(databaseSchema, "\\.")[[1]]
+    
+  } else if (is(connection, "PqConnection") || is(connection, "RedshiftConnection" || is(connection, "duckdb_connection"))) {
+    stopifnot(length(databaseSchema) == 1)
+    sql <- paste0("SELECT table_name FROM information_schema.tables WHERE table_schema = ", databaseSchema, ";")
+    DBI::dbGetQuery(connection, sql)[["table_name"]]
```
Comment (schuemie) [inline R/ListTables.R:115]: Just curious: do these drivers not support dbListTabes?

**Comment by schuemie** [comment]
I'm looking at where `dbListTables` and `getTableNames` still differ. One important distinction you introduced is that `getTableNames()` has the `databaseSchema` argument, which is the standard across HADES, whereas `dbListTables()` uses separate `database` and `schema` arguments, which to my knowledge isn't a standard anywhere. Why not also use the `databaseSchema` argument in `dbListTables()`?

File: R/ListTables.R (hunk 1 above)
Comment (ablack3) [reply R/ListTables.R:115]: They do but the differences are I how the schema are handled. If schema is null the. I just call dbListTables.

**Comment by ablack3** [comment]
> I'm looking at where `dbListTables` and `getTableNames` still differ. One important distinction you introduced is that `getTableNames()` has the `databaseSchema` argument, which is the standard across HADES, whereas `dbListTables()` uses separate `database` and `schema` arguments, which to my knowledge isn't a standard anywhere. Why not also use the `databaseSchema` argument in `dbListTables()`?

I didn’t want to change the signature of dbListTables which would be a breaking change. I think you’re right that the schema argument is not standardized in DBI although a few other backends have a “schema” argument. DBI functions use “catalog” to refer to the “database” which has only been relevant for me when using sql server. odbc uses “schema_name” and “catalog_name”. What would you prefer? “databaseScema”?

I’ve noticed table names in several database systems can have periods in them. I’m not sure if periods are allowed in schema names. If so then something like “db.schm.tbl” is ambiguous.

**Comment by schuemie** [comment]
> I didn’t want to change the signature of dbListTables which would be a breaking change. 

I think we're talking about breaking changes anyway, so I think that is ok.

I would prefer we standardize to `databaseSchema`. I suspect you're right that sometimes database, schema, and table names can have periods in them. It seems we only would have an issue if the schema name has a period, in which case we'd think the first part is the database. I haven't seen that in the wild, but we could handle that by requiring quotes or brackets around the name, so `[db.scm].tbl`. I don't think we currently handle that. We could, this also seems a very extreme edge case I wouldn't prioritize.

File: R/ListTables.R (hunk 1 above)
Comment (schuemie) [reply R/ListTables.R:115]: Ok, makes sense!

**Comment by ablack3** [comment]
@schuemie, tests have been added and are passing locally. I'm not running them on github actions. Not sure if that is necessary.

**Comment by ablack3** [comment]
Tests for getTableNames using external DBI drivers have been added. 
dbListTables has been updated so that it has a single databaseSchema argument.
I added an argument to getTableNames to allow users to opt out of the "always return uppercase" behavior since sometimes I need to see what the actual casing is on case sensitive database systems. The default behavior is unchanged so it does not break existing code. Let me know what you think.

**Review by schuemie — APPROVED** [review]

**Comment by schuemie** [comment]
Great work!
